json_encoder_test.go

```diff
@@ -66,6 +88,7 @@ func TestJSONAddString(t *testing.T) {
 
 func TestJSONAddBool(t *testing.T) {
 	withJSONEncoder(func(enc *jsonEncoder) {
+		enc.AddString("foo", "bar")
```

Comment:
No, that case is checked immediately after this assertion; the same is true for all the other AddFoo methods (any lacunae are bugs).

Nevertheless, I'll see about writing a good helper for this. It's a bit challenging to get the types right, since this code doesn't deal in Fields at all.